File: test-junit5/src/test/java/io/micronaut/test/junit5/ArgumentInjectionTest.java

package io.micronaut.test.junit5;

import io.micronaut.context.annotation.Property;
import io.micronaut.context.annotation.Value;
import io.micronaut.http.client.HttpClient;
import io.micronaut.http.client.annotation.Client;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;

@MicronautTest
@Property(name = "foo.bar", value = "test")
public class ArgumentInjectionTest {

    @Test
    void testArgumentInjected(
            MathService mathService,
            @Property(name="foo.bar") String val,
            @Client("/") HttpClient client) {
        final int result = mathService.compute(2);

        Assertions.assertEquals(8, result);
        assertNotNull(client);

        assertEquals("test", val);
    }

    @Test
    void testValueArgumentInjected(@Value("${foo.bar}") String val) {
        assertEquals("test", val);
    }

    @Test
    void testExpressionValueArgumentInjected(@Value("#{1 + 1}") Integer val) {
        assertEquals(2, val);
    }
}

File: test-suite-python/src/test/python/micronaut/test/python/ArgumentInjectionTest.py
from typing import Annotated

from micronaut.context.annotation import Property, Value
from micronaut.http.client import HttpClient
from micronaut.http.client.annotation import Client
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test

from .MathService import MathService


@MicronautTest
@Property(name="foo.bar", value="test")
class ArgumentInjectionTest:

    @Test
    def test_argument_injected(
        self,
        math_service: MathService,
        val: Annotated[str, Property(name="foo.bar")],
        client: Annotated[HttpClient, Client("/")],
    ):
        result = math_service.compute(2)

        assert 8 == result
        assert client is not None
        assert "test" == val

    @Test
    def test_value_argument_injected(self, val: Annotated[str, Value("${foo.bar}")]):
        assert "test" == val
